h/accounts/models.py

```diff
@@ -223,9 +217,6 @@ def get_by_userid(cls, domain, userid):
             # userid didn't match the pattern that split_user() expects.
             return None
 
-        if userdomain != domain:
```

Comment:
The thing is, if we're not paying any attention to the domain, I don't think we should have a `get_by_userid` at all, because it has rather misleading behaviour -- `models.User.get_by_userid('acct:foo@microsoft.com')` will happily return the user whose userid is `acct:foo@hypothes.is`.

This isn't necessarily a reason to block merging this PR. Just flagging something that needs cleaning up.